src/brevitas/graph/calibrate.py

```diff
@@ -57,9 +57,28 @@ def __enter__(self):
             self.disable_quant_inference.apply(self.model, is_training=True, quantization_enabled=False)
 
     def __exit__(self, type, value, traceback):
+        self.model.apply(finalize_collect_stats)
         self.disable_quant_inference.apply(self.model, is_training=self.previous_training_state, quantization_enabled=True)
 
 
+class bias_correction_mode():
+    def __init__(self, model, enabled=True):
+        self.model = model
+        self.call_pointer = self.model.forward
+        self.current_state = model.training
+        self.bias_correction = BiasCorrection()
+        self.enabled=enabled
+
+    def __enter__(self):
+        if self.enabled:
+            call_wrapper = partial(self.bias_correction.call_wrapper, model=self.model, wrapped_call=self.model.forward)
```

Comment:
why wrapped_call is self.model.forward and not self.model.__call__?

Reply:
1. My understanding is that \_\_call\_\_ is responsible to handle all the hooks. If we would override that, we wouldn't be able to catch all the hooks that we use for collecting FP/Quant values for then applying bias-correction.
2. It seems that torch doesn't even allow a user to change \_\_call\_\_ function, at least from my experiments it seems that just replacing forward with \_\_call\_\_ is not enough to obtain the desired behaviour.